devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+/**
+ *  Copyright 2016 SmartThings, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  Fidure Thermostat, Based on ZigBee thermostat (SmartThings)
+ *
+ *	Author: Fidure
+ *	Date: 2014-12-13
+ *  Updated: 2017-7-19
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition (
+		name: "Fidure 2nd Generation Thermostat",
+		namespace: "smartthings/tile-ux",
+		author: "SmartThings/Fidure") {
+
+		capability "Thermostat"
+		capability "Relative Humidity Measurement"
+		
+		capability "Temperature Measurement"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Sensor"
+		capability "lock" // for locking the keypad on the device.  
+		
+		command "tempUp"
+		command "tempDown"
+		command "heatUp"
+		command "heatDown"
+		command "coolUp"
+		command "coolDown"
+		command "setTemperature", ["number"]
+		command "poll"
+
+		command "setThermostatTime"
+		command "setCoolLimit", ["number"]
+		command "setHeatLimit", ["number"]
+		command "setLockLevel", ["number"]
+		command "turnProgOn"
+		command "turnProgOff"
+		command "turnHoldOn"
+		command "turnHoldOff"
+
+		fingerprint profileId: "0104", inClusters: "0000,0003,0004,0005,0201,0204,0400,0402,0405,0B05", outClusters: "000A, 0019"
+		
+		attribute "runningMode", "string"
+		attribute "setpointHold", "string"
+		attribute "prorgammingOperation", "string"
+		attribute "setpointHoldDuration", "number"
+		attribute "lockLevel", "number"
+		attribute "coolingDemand", "number"
+		attribute "heatingDemand", "number"
+		attribute "modelIdentifier", "string"
+		attribute "lastSeen","string"
+
+	}
+
+	preferences {
+		input ("lock_level", "enum", title: "Thermostat Keypad Lock Level", description: "Setting the Lock Level allows you to restric the on-device access.", options: ["Mode Only", "Setpoint","Full"], defaultValue: "Full")
+ 	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"thermostatFull", type:"thermostat", width:6, height:4) {
+			tileAttribute("device.temperature", key: "PRIMARY_CONTROL") {
+				attributeState("temp", label:'${currentValue}', unit:"dF", defaultState: true)
+			}
+			tileAttribute("device.temperature", key: "VALUE_CONTROL") {
```

Comment:
I assume this should be thermostatSetpoint, the current setpoint value which could be either the heating or cooling setpoint

Reply:
yes.  corrected.